partiql-conformance-test-generator/src/generator.rs

```diff
@@ -112,19 +250,53 @@ impl Generator {
         }
     }
 
-    fn gen_envs(&mut self, _scope: &mut Scope, _envs: &Environments) {
-        // TODO
+    fn gen_envs(&mut self, scope: &mut Scope, envs: &Environments) {
+        let env_file = self
+            .curr_mod_path
+            .iter()
+            .map(|s| s.escape_path())
+            .collect::<Vec<_>>()
+            .join("___")
+            + ".env.ion";
+
+        scope.raw(
+            quote! {
```

Comment:
Just curious, was using `quote` easier to perform certain code generation than the `codegen` crate?

Reply:
Yes. I originally started using it when trying to create a [OnceCell](https://docs.rs/once_cell/latest/once_cell/) to hold the 'envs' as a lazily implemented global static variable, as it was much clearer to do with `quote`.

It looked something like:

```rust

let tokens = quote! {
    const ENV_ION_TEXT: &'static str = #ion_text_str;
    static ENV_STRUCT: once_cell::unsync::Lazy<Mutex<Struct>> = Lazy::new(|| {
        let element_stream = element_reader().read_all("foo".as_bytes()).expect("ion value");
        let element = element_stream[0];
        element.as_struct().expect("struct")
    });
};

```

But, the currently implementation of ion-rust's `Element` uses an `Rc`, which makes it neither `Send` nor `Sync`, so I was unable to create the static variable.


Another way to think about it is: `codegen` makes it convenient to create and pass around structure (e.g., `scope`s, `module`s, `fn`s), but it doesn't provide any particular functionality for lines of logic; `quote` is nice for this.